### DISCUSSION TITLE
Firmware on the ESP32-PICO-KIT over USB

### DISCUSSION THREAD

**Discussion by Mr-JimmyDeath** [thehighnibble/thehighnibble.github.io#133] [IMSAI8080esp replica]
Hello

The build instructions talk about a couple of wires that need to be fitted to the ESP32 in order to enable the subject feature however the build instructions simply says 'details to follow', can I have the instructions please.

I'm aware that you can also update over WiFi however as I am building this for someone else I want to ensure that all the features are in place before I finish it.

Thanks, Regards James

**Comment by Mr-JimmyDeath** [comment]
Shame no one has an answer to this as I shall have to return the boards to their new owner without the mod.

Here are some other observations that you may like to incorporate into the instructions:

IMSAI 8080
1. If you choose to fit Q1 ensure that is sits no higher than the shorter of the two nylon spacers otherwise it will foul the rear perspex panel. (It is a real pain to disassemble the case at the point you realise it is too tall and there were no cut outs for it)
2. At the Testing 1 stage it would be useful if the instructions said what the Wifi password was rather than making the constructor search for it.

S-132
1-It was not obvious to me that flashing the firmware had to be done via the FTDI pins on the S-132. I actually started by trying to use the serial ports on the back of the 8080 (doh!). This was on the basis that the flasher tool was looking for a serial port.
2-Flashing only worked for me with the S-132 detached from the 8080.
3-If you power up the S-132 not connected to the 8080 it outputs a series of dots on the VGA port (this might be a way of showing that it is working)
4- Might be worth pointing out that in the configuration file to enable a line you need to remove the # symbol. I initially edited the lines to add the S-132 but didn't spot that the # had been removed. Took me ages to work out that even when fitted to the 8080 why it only printed out dots.

Regards

James

**Reply by dmcnaugh** [reply]
> 1-It was not obvious to me that flashing the firmware had to be done via the FTDI pins on the S-132.

The ESP32 module for the S-132 is supplied with the firmware already flashed on. No flashing has to be done.

> 2-Flashing only worked for me with the S-132 detached from the 8080.

Yes, makes sense as the Rx/Tx lines are connected to the IMSAI when attached.

> 3-If you power up the S-132 not connected to the 8080 it outputs a series of dots on the VGA port

Yes, this is the S-132 waiting on the handshake with the IMSAI8080

>  4- Might be worth pointing out that in the configuration file to enable a line you need to remove the # symbol.

Lines starting with `#` are comments, that's fairly conventional for many config file formats

**Reply by Mr-JimmyDeath** [reply]
"The ESP32 module for the S-132 is supplied with the firmware already flashed on". The web page goes from build notes to firmware install so that's what I did. I didn't see any statement about the ESP being pre-programmed.

Something else that just popped to mind that may be of use to future builders is a note in the 8080 build before completion of the case that if they have the S-132 option they may want to drill the mounting hole. A printable template would be useful here.

**Comment by dmcnaugh** [comment]
> The build instructions talk about a couple of wires that need to be fitted to the ESP32

This was a dead-end feature that is not required and should be removed from the documentation.
You are the first person in 4 years to even make note of it.
If the ESP32 module needs to be re-flashed via USB then it simply needs to be removed from the IMSAI8080 PCB first.

**Reply by Mr-JimmyDeath** [reply]
Ah ok, that will save having to ensure the correct height of Q1. The way it is written suggests that it is a feature to come, It's removal would have saved me having to strip down the case to adjust the height if I had known.
